integration_tests/src/main/python/cache_test.py

```diff
@@ -152,7 +154,7 @@ def op_df(spark, length=2048, seed=0):
 
     assert_gpu_and_cpu_are_equal_collect(op_df, conf = enableVectorizedConf)
 
-@pytest.mark.parametrize('data_gen', all_gen, ids=idfn)
+@pytest.mark.parametrize('data_gen', all_gen + [all_basic_struct_gen, StructGen([['child0', byte_gen]])], ids=idfn)
```

Comment:
What about testing structs containing more than one type or deeper nesting such as struct-of-structs?

Reply:
@jlowe I have updated the tests for testing nested struct. PTAL